src/controller/CommissioningDelegate.h

```diff
@@ -762,6 +779,9 @@ struct ICDManagementClusterInfo
 
 struct ReadCommissioningInfo
 {
+#if CHIP_CONFIG_ENABLE_READ_CLIENT
```

Comment:
Agree with the comments above - nothing in the Commissioner reasonably works without a read client. I think this is mostly a remnant from an ambitious flagging effort right near the start of the project that wasn't properly maintainted. If this continues to get in your way, ripping it out is probably the way to go. If someone really wants to write a non-commissioning, command only controller, this define is not going to save them.
It does need to be on, but could just do a define guard with an error if it's not on when the controller is compiled in.

Anyway, not important for this particular PR, just thought I'd put that out there. Feel free to close this comment.